src/Npgsql/NpgsqlParameter.cs

```diff
@@ -278,10 +277,9 @@ internal void ChangeParameterName(string? value)
         get => _value;
         set
         {
-            if (_value == null || value == null || _value.GetType() != value.GetType())
-                Handler = null;
+            if (value == null || _value?.GetType() != value.GetType())
```

Comment:
Operator methods like `==` (op_Equality) are always resolved on the static type, they are not virtual. As object doesn't have any it then falls back to emitting `ceq` (this would also happen for int, or even T)

Still good to move to more pattern matching though.

Reply:
Addressed